fs/accounting/stats_groups.go

```diff
@@ -375,6 +375,9 @@ func (sg *statsGroups) sum(ctx context.Context) *StatsInfo {
 			sum.startedTransfers = append(sum.startedTransfers, stats.startedTransfers...)
 			sum.oldDuration += stats.oldDuration
 			sum.oldTimeRanges = append(sum.oldTimeRanges, stats.oldTimeRanges...)
+			stats.average.mu.Lock()
+			defer stats.average.mu.Unlock()
```

Comment:
Can you do the locking without `defer` as there may be lots of `sg.m` and we don't need to keep everything locked until the end of the function, so

```
stats.average.mu.Lock()
sum.average.speed += stats.average.speed
stats.average.mu.Unlock()
```

I'm not happy about the stats stuff in general - it is all a bit of a mess, but I don't expect you to fix that!

Reply:
Hi @ncw Thanks for bringing that up, I think we should discuss about how stats should be provided. Currently, its very difficult to show everything in the UI. 

I think we should give everything we have about current transfers in the core/stats itself,
- Current upload speed
- Current download speed
- Currently transferring
- Checking/ Renaming/ Deleting
- Transferred

Reply:
Do you want to make an issue with what we have at the moment and what you'd like it? Then we can discuss - thank you :-)

Reply:
Yes, I will do that. Thanks

Reply:
https://github.com/rclone/rclone/issues/5534